c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -510,6 +581,46 @@ class DeviceCachingAllocator {
     }
   }
 
+  // CUDAGraph interactions (see Note [Interaction with CUDA graph capture])
+
+  // Called by CUDAGraph::capture_begin
+  void notifyCaptureBegin(CUDACaptureid_t graph_id, CUDACaptureid_t mempool_id) {
```

Comment:
Probably some sort of invariant here that `mempool_id` must not already be mapped in `capture_to_pool_map`?

Reply:
Aye, that's the assert you recommended i can streamline https://github.com/pytorch/pytorch/pull/51436/files#r568255599